### PR TITLE
Replace terraform-plugin-sdk with terraform-plugin-testing

### PR DESCRIPTION
**Pull request by trodge**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/acctest/go/resource_test_utils.go
Comment (zli82016) [inline mmv1/third_party/terraform/acctest/go/resource_test_utils.go:1]: This file is needed. `go` folder is for the converted files from .go.erb files. Sorry for the confusion.

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/acctest/go/resource_test_utils.go
Comment (zli82016) [reply mmv1/third_party/terraform/acctest/go/resource_test_utils.go:1]: This file can be removed.

**Comment by zli82016** [comment]
Do you mind resolving the conflicts, @trodge ? The changes LGTM.

**Comment by zli82016** [comment]
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10339-old..auto-pr-10339) ( 1068 files changed, 1780 insertions(+), 1768 deletions(-)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10339-old..auto-pr-10339) ( 1233 files changed, 2011 insertions(+), 1999 deletions(-))
> 
> ## Errors
> `google` provider:
> 
> * The diff processor crashed while computing breaking changes. This is usually due to the downstream provider failing to compile.
> * The diff processor crashed while computing changed resource schemas.
> 
> `google-beta` provider:
> 
> * The diff processor crashed while computing breaking changes. This is usually due to the downstream provider failing to compile.
> * The missing test detector failed to run.
> * The diff processor crashed while computing changed resource schemas.

@trodge ，do you have an idea what caused the error?

**Comment by trodge** [comment]
> > Hi there, I'm the Modular magician. I've detected the following information about your changes:
> > ## Diff report
> > Your PR generated some diffs in downstreams - here they are.
> > `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10339-old..auto-pr-10339) ( 1068 files changed, 1780 insertions(+), 1768 deletions(-)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10339-old..auto-pr-10339) ( 1233 files changed, 2011 insertions(+), 1999 deletions(-))
> > ## Errors
> > `google` provider:
> > 
> > * The diff processor crashed while computing breaking changes. This is usually due to the downstream provider failing to compile.
> > * The diff processor crashed while computing changed resource schemas.
> > 
> > `google-beta` provider:
> > 
> > * The diff processor crashed while computing breaking changes. This is usually due to the downstream provider failing to compile.
> > * The missing test detector failed to run.
> > * The diff processor crashed while computing changed resource schemas.
> 
> @trodge ，do you have an idea what caused the error?

It's caused by importing `terraform-plugin-sdk/v2/helper/resource` alongside `terraform-plugin-testing/helper/resource`.

These are the only files that still import `terraform-plugin-testing/v2/helper/resource`:

```
mmv1/third_party/terraform/services/netapp/go/resource_netapp_storage_pool_test.go.tmpl
mmv1/third_party/terraform/acctest/go/framework_test_utils.go
```

Based on this, I think I need to keep the changes I reverted in the most recent commit.

**Comment by zli82016** [comment]
> Edit- after considering the logs further I've realized that the reason this happens is that the diff processor is building the old version of the provider which still uses `terraform-plugin-sdk` alongside the new version using `terraform-plugin-testing`.

Will it also cause error after merging this PR in the main branch?
I am so familiar with the diff processor. Added @roaks3 as a secondary reviewer.

**Comment by trodge** [comment]
> > Edit- after considering the logs further I've realized that the reason this happens is that the diff processor is building the old version of the provider which still uses `terraform-plugin-sdk` alongside the new version using `terraform-plugin-testing`.
> 
> Will it also cause error after merging this PR to the main branch? I am so familiar with the diff processor. Added @roaks3 as a secondary reviewer.

It shouldn't cause any errors after it's merged, since the old version using `terraform-plugin-sdk` will be gone.

**Comment by zli82016** [comment]
> > > Edit- after considering the logs further I've realized that the reason this happens is that the diff processor is building the old version of the provider which still uses `terraform-plugin-sdk` alongside the new version using `terraform-plugin-testing`.
> > 
> > 
> > Will it also cause error after merging this PR to the main branch? I am so familiar with the diff processor. Added @roaks3 as a secondary reviewer.
> 
> It shouldn't cause any errors after it's merged, since the old version using `terraform-plugin-sdk` will be gone.

That makes sense. Thanks.

**Comment by zli82016** [comment]
LGTM. Waiting for confirmation from @roaks3

**Review by roaks3 — APPROVED** [review]
Yea, the logic makes sense to me regarding the diff processor failure. It would be nice if we could verify before merging, but I don't think it would be feasible for a case like this, so we should just be ready to revert if needed (and also not merge late on a Friday).

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**
